spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -150,7 +150,7 @@ public Object postProcess(Object bean, String beanName, Method method, List<Anno
 					((Advised) handler).addAdvisor(advisor);
 				}
 				else {
-					ProxyFactory proxyFactory = new ProxyFactory(bean);
+					ProxyFactory proxyFactory = new ProxyFactory(handler);
```

Comment:
I hit it while testing `IdempotentReceiverIntegrationTests` (after I removed the JMX proxy) - since the test is in JMX, it always took the other branch.